src/waterfall/waterfall.cpp

```diff
@@ -398,4 +399,13 @@ void Waterfall::hoverLeaveEvent(QHoverEvent *event)
     Q_UNUSED(event)
     emit mouseLeave();
     _mouseDepth = -1;
+    _mouseInside = false;
+    emit mouseInsideChanged();
+}
+
+void Waterfall::hoverEnterEvent(QHoverEvent *event)
+{
+    Q_UNUSED(event)
```

Comment:
should we emit a `mouseEnter()` here?
do we need to do anything with `_mouseDepth`?